### PR TITLE
Configuration management for Concourse

### PR DESCRIPTION
**Pull request by blarghmatey**
This PR adds the initial framework for how we will be using PyInfra as well as the implementation of getting a Concourse system installed.

### CODE REVIEW

File: src/ol_configuration_management/components/baseline/setup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+@deploy("Install baseline requirements")
+def install_baseline_packages(state=None, host=None):
+    apt.packages(
+        name="Install baseline packages for Debian based hosts",
+        packages=[
+            "curl",
+        ],
+        update=True,
```
Comment (shaidar) [inline src/ol_configuration_management/components/baseline/setup.py:12]: Would it be helpful to add `cache_time=3600` or some other value to reduce repetitive requests?

File: src/ol_configuration_management/facts/has_systemd.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from pyinfra.api import FactBase
```
Comment (shaidar) [inline src/ol_configuration_management/facts/has_systemd.py:1]: Maybe adding a comment here about why this file is needed

File: src/ol_configuration_management/components/concourse/build.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        present=True,
+        user=concourse_config.user,
+        recursive=True,
+        state=state,
+        host=host,
+        sudo=sudo,
+    )
+    # Download latest Concourse release from GitHub
+    concourse_archive = f"https://github.com/concourse/concourse/releases/download/v{concourse_config.version}/concourse-{concourse_config.version}-linux-amd64.tgz"
+    concourse_archive_hash = f"https://github.com/concourse/concourse/releases/download/v{concourse_config.version}/concourse-{concourse_config.version}-linux-amd64.tgz.sha1"
```
Comment (shaidar) [inline src/ol_configuration_management/components/concourse/build.py]: Should the URLs live in the model?

File: src/ol_configuration_management/deployments/concourse.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+from ol_configuration_management.components.concourse.models import (
+    ConcourseBaseConfig,
+    ConcourseWebConfig,
+    ConcourseWorkerConfig,
+)
+from ol_configuration_management.facts import has_systemd  # noqa: F401
+
+concourse_config = ConcourseBaseConfig()
+web_config = ConcourseWebConfig()
+worker_config = ConcourseWorkerConfig()
```
Comment (shaidar) [inline src/ol_configuration_management/deployments/concourse.py]: Why is this defined here but not being used?

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  # config.vm.provision "shell", inline: <<-SHELL
+  #   apt-get update
+  #   apt-get install -y apache2
+  # SHELL
```
Comment (shaidar) [inline Vagrantfile:79]: Any reason/use to keep all those commented lines?

**Comment by shaidar** [comment]
Overall looks good as far as I can tell. I'd imagine once you run the entire workflow you'll notice something here and there, but it's 👍 for me

File: Vagrantfile (hunk 5 above)
Comment (blarghmatey) [reply Vagrantfile:79]: No particular reason.

File: src/ol_configuration_management/deployments/concourse.py (hunk 4 above)
Comment (blarghmatey) [reply src/ol_configuration_management/deployments/concourse.py]: Just haven't integrated it yet. The plan is to add a switch to determine whether to pull in the config for a worker or a web node at build time and propagate the appropriate config.

**Pull request merged**